pkg/provision/providers/qemu.go

```diff
@@ -2,17 +2,26 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-//go:build linux
+//go:build linux || darwin
 
 package providers
 
 import (
 	"context"
+	"errors"
+	"os"
+	"runtime"
 
 	"github.com/siderolabs/talos/pkg/provision"
 	"github.com/siderolabs/talos/pkg/provision/providers/qemu"
 )
 
 func newQemu(ctx context.Context) (provision.Provisioner, error) {
+	// TODO: remove once the provisioner works on darwin
+	_, darwinQemuEnvIsSet := os.LookupEnv("DARWIN_QEMU")
```

Comment:
I think we can skip this for now, as the code panics on Darwin, it's good enough for `main` (as we're WIP on adding Darwin support)